src/schemas/schema_validator.ts

```diff
@@ -21,13 +22,19 @@ export class SchemaValidator {
         this._validator.addSchema(schema, schema.id);
     }
 
-    // In order to validate a complex JS object using jsonschema, we must replace any complex
-    // sub-types (e.g BigNumber) with a simpler string representation. Since BigNumber and other
-    // complex types implement the `toString` method, we can stringify the object and
-    // then parse it. The resultant object can then be checked using jsonschema.
     public validate(instance: any, schema: Schema): ValidatorResult {
-        const jsonSchemaCompatibleObject = JSON.parse(JSON.stringify(instance));
-        return this._validator.validate(jsonSchemaCompatibleObject, schema);
+        Validator.prototype.customFormats.BigNumber = function(input) {
+            const regex = RegExp("^\\d+(\\.\\d+)?$");
+            // This allows undefined inputs, e.g. salt is not present sometimes.
```

Comment:
This feels like something a bit code-smellish that shouldn't be necessary -- why are format validations getting run on optional parameters?

Reply:
If this is something unavoidable I'm okay with it, but my impression was that json schema validations _in general_ don't run on optional parameters unless they're present.